File: chapters/01_gateway_event_loop/v2_envelope.ts
// TypeScript mirror for Chapter 01: Gateway Event Loop (v2)
// Run with: npx ts-node v2_envelope.ts

interface Envelope {
  id: string;
  channel: string;
  user: string;
  payload: { text: string };
  meta: { version: number };
}

function buildEnvelope(channel: string, user: string, text: string): Envelope {
  return {
    id: `${channel}:${user}:001`,
    channel,
    user,
    payload: { text },
    meta: { version: 2 },
  };
}

function classifyIntent(text: string): string {
  const lowered = text.toLowerCase();
  if (lowered.includes("create") || text.includes("创建")) {
    return "create_file";
  }
  if (lowered.includes("read") || text.includes("读取")) {
    return "read_file";
  }
  return "reply";
}

function handleEnvelope(envelope: Envelope): Record<string, string> {
  const text = envelope.payload?.text ?? "";
  const intent = classifyIntent(text);
  return {
    id: envelope.id,
    intent,
    result: `handled:${intent}`,
  };
}

function main(): void {
  const env = buildEnvelope("telegram", "u1", "please create file");
  console.log(handleEnvelope(env));
}

main();

File: chapters/01_gateway_event_loop/v2_envelope.py
from __future__ import annotations


def build_envelope(channel: str, user: str, text: str) -> dict[str, object]:
    return {
        "id": f"{channel}:{user}:001",
        "channel": channel,
        "user": user,
        "payload": {"text": text},
        "meta": {"version": 2},
    }


def classify_intent(text: str) -> str:
    lowered = text.lower()
    if "create" in lowered or "创建" in text:
        return "create_file"
    if "read" in lowered or "读取" in text:
        return "read_file"
    return "reply"


def handle_envelope(envelope: dict[str, object]) -> dict[str, object]:
    payload = envelope.get("payload", {})
    text = payload.get("text", "") if isinstance(payload, dict) else ""
    intent = classify_intent(str(text))
    return {
        "id": envelope.get("id", "unknown"),
        "intent": intent,
        "result": f"handled:{intent}",
    }


def main() -> None:
    env = build_envelope("telegram", "u1", "please create file")
    print(handle_envelope(env))


if __name__ == "__main__":
    main()
